qiskit/pulse/schedule.py

```diff
@@ -814,69 +819,186 @@ def wrapper(self, *args, **kwargs):
+    Now you assign the actual pulse program to this reference.
 
-    The ``ScheduleBlock`` defaults to the ``AlignLeft`` alignment.
-    The timing overlap constraint of instructions is not immediately evaluated,
-    and thus we can assign a parameter object to the instruction duration.
-    Instructions are implicitly scheduled at optimum time when the program is executed.
+    .. jupyter-execute::
 
-    Note that ``ScheduleBlock`` can contain :class:`~qiskit.pulse.instructions.Instruction`
-    and other ``ScheduleBlock`` to build an experimental program, but ``Schedule`` is not
-    supported. This should be added as a :class:`~qiskit.pulse.instructions.Call` instruction.
-    This conversion is automatically performed with the pulse builder.
+        sched2.assign_references({"grand_child": sched1})
+        print(sched2.scoped_parameters)
 
-    By using ``ScheduleBlock`` representation we can fully parametrize pulse waveforms.
-    For example, Rabi schedule generator can be defined as
+    Now you get two parameters "root.amp" and "root.grand_child.amp".
+    The second parameter indicates it is defined within the referred program "grand_child".
+    The program calling the "grand_child" has a reference program description
+    which is accessed through :attr:`ScheduleBlock.references`.
 
-    .. code-block:: python
+    .. jupyter-execute::
 
-        duration = Parameter('rabi_dur')
-        amp = Parameter('rabi_amp')
+        print(sched2.references)
 
-        block = ScheduleBlock()
-        rabi_pulse = pulse.Gaussian(duration=duration, amp=amp, sigma=duration/4)
+    Finally, you may want to call this program from another program.
+    Here we try different approach to define subroutine, namely, we call
+    a subroutine from the root program with actual program ``sched2``.
 
-        block += Play(rabi_pulse, pulse.DriveChannel(0))
-        block += Call(measure_schedule)
+    .. jupyter-execute::
 
-    Note that such waveform cannot be appended to the ``Schedule`` representation.
+        amp3 = Parameter("amp")
 
-    In the block representation, the interval between two instructions can be
-    managed with the ``Delay`` instruction. Because the schedule block lacks an instruction
-    start time ``t0``, we cannot ``insert`` or ``shift`` the target instruction.
-    In addition, stored instructions are not interchangable because the schedule block is
-    sensitive to the relative position of instructions.
-    Apart from these differences, the block representation can provide compatible
-    functionality with ``Schedule`` representation.
+        with pulse.build() as main:
+            pulse.play(pulse.Constant(300, amp3), pulse.DriveChannel(0))
+            pulse.call(sched2, name="child")
+
+        print(main.scoped_parameters)
+
+    This implicitly creates a reference within the root program and assign ``sched2`` to it.
+    You get three parameters "root.amp", "root.child.amp", and "root.child.grand_child.amp".
```

Comment:
This needs to be adjusted to the delimiter, i.e. `:`, no? Or perhaps `::` as @wshanks suggests.

Reply:
Yes, thanks for catching this :)